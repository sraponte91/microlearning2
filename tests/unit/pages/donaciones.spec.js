import { config, createLocalVue, mount } from '@vue/test-utils'
import VueMeta from 'vue-meta'
import Donaciones from '@/pages/donaciones.vue'
// create vue with vue-meta
const localVue = createLocalVue()
localVue.use(VueMeta, { keyName: 'head' })

const mock = {
    localVue,
    mocks: {}
}

const wrapper = mount(Donaciones, mock)

describe('donaciones.vue', () => {
    test('Is a Vue instance', () => {
        expect(wrapper.vm).toBeTruthy()
    })
    test('Renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    test('has correct <head> content', () => {
        expect(wrapper.vm.$metaInfo.title).toBe(config.mocks['$constants'].donaciones.meta.title)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'title'
        ).content).toBe(config.mocks['$constants'].donaciones.meta.metaTitle)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'description'
        ).content).toBe(config.mocks['$constants'].donaciones.meta.metaDescription)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'og:title'
        ).content).toBe(config.mocks['$constants'].donaciones.meta.ogTitle)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'og:description'
        ).content).toBe(config.mocks['$constants'].donaciones.meta.ogDescription)

        expect(wrapper.vm.$metaInfo.meta.find(
            (item) => item.hid === 'og:image'
        ).content).toBe(config.mocks['$constants'].baseURL + config.mocks['$constants'].donaciones.meta.ogImage)
    })
})
