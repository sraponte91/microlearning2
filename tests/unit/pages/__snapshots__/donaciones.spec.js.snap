// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`donaciones.vue Renders correctly 1`] = `
<main>
  <section
    class="main-slider-at-top donaciones-main position-relative"
  >
    <div
      class="img-prt position-relative"
    >
      <img
        alt=""
        class="object-fit-cover"
        src="/images/head_fe_8241301e_6_afbc_1_272402.jpeg"
      />
    </div>
     
    <div
      class="container position-absolute"
    >
      <h1
        class="text-uppercase ff-sans-b"
      >
        
        Su Donación Tiene El 
        <br />
        
        Poder De Transformar 
        <br />
        
        Historia
      
      </h1>
       
      <div
        class="iwant-donate"
      >
        <a
          class="ff-nunito btn-iwant-do"
          href="https://www.zonapagos.com/formulario/"
          target="_blank"
        >
          Quiero donar
        </a>
      </div>
    </div>
  </section>
   
  <section
    class="bg-white donate-funding"
  >
    <div
      class="container"
    >
      <div
        class="inner-wid margin-0-auto"
      >
        <div
          class="img-part position-relative after-po"
        >
          <img
            alt=""
            class="object-fit-cover"
            src="/images/ninios_campana_8c3f2bb2d28185797bbc6f.jpeg"
          />
           
          <div
            class="position-absolute set-btom text-center"
          >
            <h3
              class="text-uppercase ff-sans-b"
            >
              
              Transformando El Futuro De Nuestros Niños
            
            </h3>
          </div>
        </div>
         
        <div
          class="progress-bar-here"
        >
          <div
            class="text-right"
          >
            <p>
              Recaudado
            </p>
          </div>
           
          <div
            class="progress"
          >
            <div
              aria-valuemax="50"
              aria-valuemin="0"
              aria-valuenow="33"
              class="progress-bar"
              role="progressbar"
              style="width: 66.666666666%;"
            />
          </div>
           
          <h4>
            $4.192.000 de $14.000.000
          </h4>
        </div>
         
        <div
          class="text-info-ab-do ff-nunito"
        >
          <p>
            
            En Medellín más de 150 niños y niñas en situación de abandono, han
            sido acogidos por la Corporación Superarse. Esta organización los
            apoya en la construcción de su proyecto de vida y los forma en
            habilidades tecnológicas para acceder a su derecho a la educación.
            Hoy se hace imprescindible disponer de computadores como
            herramienta de aprendizaje y esto será posible gracias a su
            donación. Apoyemos este camino de esperanza.
          
          </p>
        </div>
         
        <div
          class="donate-btn text-center"
        >
          <a
            class="btb-do-b ff-nunito"
            href="https://www.zonapagos.com/formulario/"
            target="_blank"
          >
            Quiero donar
          </a>
        </div>
      </div>
    </div>
  </section>
   
  <section
    class="be-part-of-ther position-relative"
  >
    <img
      alt=""
      class="object-fit-cover d-none-767"
      src="/images/main_rectangle_abde.svg"
    />
     
    <div
      class="container position-absolute text-btn-lel position-relative-767"
    >
      <div
        class="inn-sm-bx margin-0-auto"
      >
        <div
          class="d-flex justify-content-center flex-wrap-767"
        >
          <div
            class="text-right txt-part text-center-767"
          >
            <h3
              class="text-uppercase ff-sans-b"
            >
              <span>
                sea Parte
              </span>
              <br />
              
              De Los Que
              <br />
              
              Transforman
              <br />
              
              Historias
            
            </h3>
          </div>
           
          <div
            class="nd-part-of-div"
          >
            <p
              class="ff-nunito"
            >
              
              Con una donación usted tiene el poder de transformar la historia
              de estas iniciativas, brindándoles más oportunidades y recursos
              para que sigan construyendo el futuro del país.
            
            </p>
             
            <div
              class="donate-btn"
            >
              <a
                class="btb-do-b ff-nunito"
                href="https://www.zonapagos.com/formulario/"
                target="_blank"
              >
                Quiero donar
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="seprate-iimg position-absolute position-relative-767"
    >
      <img
        alt=""
        class="object-fit-cover"
        src="/images/background278a6_bc9a61e873ed.png"
      />
    </div>
  </section>
   
  <section
    class="joing-forcese"
  >
    <div
      class="container"
    >
      <div
        class="inn-sm-sec margin-0-auto"
      >
        <h3
          class="text-uppercase ff-sans-b"
        >
          Unimos Fuerzas
        </h3>
         
        <div
          class="text-center pra-p"
        >
          <p
            class="ff-nunito"
          >
            
            Estos son todos los aliados que se han unido para ayudar a las
            comunidades, líderes y emprendedores a transformar sus vidas e
            iniciativas.
          
          </p>
        </div>
      </div>
    </div>
  </section>
   
  <section
    class="donation-certificate bg-white"
  >
    <div
      class="container"
    >
      <div
        class="text-center"
      >
        <div
          class="img margin-0-auto"
        >
          <img
            alt=""
            class="object-fit-cover"
            src="/images/superarse_095a4fdf_af0b3e-482_e4572e88f2_a.jpeg"
          />
        </div>
         
        <div
          class="pra-p ff-nunito"
        >
          <p>
            Para solicitar su certificado de donación contacte a
          </p>
           
          <a
            href="mailto:donaciones@fundacionbd.org"
          >
            donaciones@fundacionbd.org
          </a>
        </div>
      </div>
    </div>
  </section>
</main>
`;
